### PR TITLE
Mark some more types as `UnwindSafe`

### PR DESCRIPTION
**Pull request by tbu-**
Mark `Atomic*` `UnwindSafe`: It's the same in the standard library.

Mark everything in `tokio::sync::mpsc` as `UnwindSafe`: Since these types are safe in the face of panics and cannot cause data corruption, they're `UnwindSafe` irrespective of the channel's element type.

CC #4657.

### CODE REVIEW

**Comment by tbu-** [comment]
I can do the rest of the channels if desired.

File: tokio/tests/sync_mpsc.rs [hunk 1]
```diff
@@ -23,8 +24,36 @@ mod support {
 
 #[allow(unused)]
 trait AssertSend: Send {}
-impl AssertSend for mpsc::Sender<i32> {}
 impl AssertSend for mpsc::Receiver<i32> {}
+impl AssertSend for mpsc::Sender<i32> {}
+impl AssertSend for mpsc::UnboundedReceiver<i32> {}
+impl AssertSend for mpsc::UnboundedSender<i32> {}
+impl AssertSend for mpsc::WeakSender<i32> {}
+impl AssertSend for mpsc::WeakUnboundedSender<i32> {}
```
Comment (Darksonn) [inline tokio/tests/sync_mpsc.rs:32]: There's some overlap here with this file. I'd rather make sure the list is complete here.

https://github.com/tokio-rs/tokio/blob/365269adaf6ec75743c0693f2378c3c6d04f806b/tokio/tests/async_send_sync.rs#L398-L415

**Review by Darksonn — APPROVED** [review]
Overall looks good to me.

File: tokio/tests/sync_mpsc.rs (hunk 1 above)
Comment (tbu-) [reply tokio/tests/sync_mpsc.rs]: Fixed. Moved all the `Send` tests there, added the missing ones.

**Review by Darksonn — APPROVED** [review]
Thanks.

**Pull request merged**